packages-web/maps/src/components/MapsContainer.ts

```diff
@@ -220,16 +220,18 @@ class MapsContainer extends Component<MapsContainerProps, MapsContainerState> {
         );
     }
 
-    private onClickMarker = (event: LeafletEvent & google.maps.MouseEvent, locationAttr: DataSourceLocationProps) => {
+    private onClickMarker = (event: LeafletEvent & GoogleMapsClickEvent, locationAttr: DataSourceLocationProps) => {
         const { locations } = this.state;
-        const latitude = this.props.mapProvider === "googleMaps" ? event.latLng.lat() : event.target.getLatLng().lat;
-        const longitude = this.props.mapProvider === "googleMaps" ? event.latLng.lng() : event.target.getLatLng().lng;
-        this.executeAction(
-            locations[
-                locations.findIndex(targetLoc => targetLoc.latitude === latitude && targetLoc.longitude === longitude)
-            ],
-            locationAttr
-        );
+        const GUID = this.props.mapProvider === "googleMaps" ? event.options.GUID : event.target.options.GUID;
+
+        if (GUID) {
+            this.executeAction(
+                locations[
+                    locations.findIndex(targetLoc => targetLoc.mxObject && targetLoc.mxObject.getGuid() === GUID)
```

Comment:
Think we can use, if we can already use `Array.prototype.find()` not sure if it polyfilled for IE11

Reply:
Yes, I tested and it works perfectly. Just pushed the fix